resources/git-fork.md

```diff
@@ -0,0 +1,98 @@
+* **Clean main repository:** as in the above case, but even for smaller teams, often the main repository is kept as the source-of-truth for the `master` (or `develop`) branch, and is otherwise reserve for CI and CD tasks, such as preserving production release versions. Branches for work-in-progress are relegated to user forks only in order to keep the central repo tidy.
+* **Experimental changes:** Sometimes you need to make large architectural changes or adjust CI/CD configuration, and it isn't acceptable to block work on the main repo. Configuring another pipeline against a fork is one option to get around this.
+
+## Dealing with multiple `remote`s
+When you set up your local repository with multiple `remote` repositories (usually the main repository and your own fork), there are slight changes to some of the `git` commands you are accustomed to using. Specifically, for any commands that deal with remote repositories (`push`, `pull`, `fetch` etc), you now need to specify the repository with which you want to interact.
+
+For the following examples, imagine you cloned from your fork, so the repo copy under your account is the remote named `origin`, and we've added a second remote called `upstream` for the main reposiory.
+
+### `pull`
+When pulling the latest changes for a tracked remote branch, you now specifiy the full source, including the remote name:
+```shell
+$> git pull upstream master
+```
+
+### `push`
+Now when you push a branch, you will nee to specifiy the target remote by name:
+```shell
+$> git push origin my-cool-feature
+```
+A common workflow is to push your work-in-progress feature branches to your fork, and then issue a Pull Request from there back to the main repository.
+
+### `fetch`
+If you want to retrieve all of the latest changes from some remote (usually `upstream`), you can do so via `fetch`:
+```shell
+$> git fetch upstream
+```
+
+### `rebase`
+_TBD_ - should we cover this?
+
+## The Short Version
+If your team uses forks as part of their flow, after get acess to the main repository you should do the following:
+1) Create a personal fork of the repository on whichever platform the team uses.
+1) Clone _your fork_ locally: `git clone git@github.com:your-account/shared-repo.git`
+1) Add the main repository as another remote: `git remote add upstream git@github.com:repo-owner-account/shared-repo.git`
+1) Checkout the `master` branch
+1) Create a branch for your feature: `git checkout -b feat/my-cool-feature`
+1) Develop! Commit! Get it to the point where you're ready to propose merging the code.
+1) Get the latest code from the upstream repo (other code may have been merged): `git fetch upstream`
+1) Rebase your branch against the latest version of `master` from `upstream`: `git rebase upstream/master`
+  1) Follow the steps, being very carefully resolve any _merge conflicts_ if necessary.
```

Comment:
typo, `being very carefully resolve` should be `being very careful to resolve`

Reply:
do we want to link to the rebase document here?

Reply:
Added. Also for branching.